fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Optimizer.java

```diff
@@ -153,14 +159,191 @@ public OptExpression optimize(ConnectContext connectContext,
 
         context.getTaskScheduler().executeTasks(rootTaskContext, memo.getRootGroup());
 
-        OptExpression result = extractBestPlan(requiredProperty, memo.getRootGroup());
+        OptExpression result;
+        if (!connectContext.getSessionVariable().isSetUseNthExecPlan()) {
+            result = extractBestPlan(requiredProperty, memo.getRootGroup());
+        } else {
+            // extract the nth execution plan
+            int nthExecPlan = connectContext.getSessionVariable().getUseNthExecPlan();
+            result = extractNthPlan(requiredProperty, memo.getRootGroup(), nthExecPlan);
+            if (result == null) {
+                throw new StarRocksPlannerException("Can not extract " + nthExecPlan + " from Memo",
+                        ErrorType.INTERNAL_ERROR);
+            }
```

Comment:
```suggestion
            if (result == null) {
                throw new StarRocksPlannerException("Can not extract " + nthExecPlan + " from Memo",
                        ErrorType.INTERNAL_ERROR);
            }
```

Reply:
?

Reply:
Don't need the check, can be throw Expection on line 184 direct